Marlin/src/module/tool_change.cpp

```diff
@@ -1036,6 +1037,10 @@ void tool_change(const uint8_t new_tool, bool no_move/*=false*/) {
 
     planner.synchronize();
 
+    #if HAS_LEVELING
+      set_bed_leveling_enabled(leveling_was_enabled);
+    #endif
+
```

Comment:
This is the only functional difference I can discern from this PR's full set of changes. Instead of restoring leveling at the end (or on multiplexer error), leveling is restored here ahead of solenoids, mmu, servo, and fanmux.

Again, it would be good to try logging, adding short messages ahead of where we expect the remembered state to get restored, and then also adding logging to the `TemporaryBedLevelingState` class. If the restore happens in an unexpected spot then we can investigate further.

Reply:
I wont have time to dig into it like that, and saving the couple lines above isnt worth the effort IMO for what should be even if working binary identical code. The express command ensures its a stateful event handled exactly when meant to because it contains motion. Its safe. Also anyone coming into the code later on wont be left wondering whats happening if they come across it. It adds to the "tribal knowledge" requirement that limits developers. If the goal at the moment is to march to RC1, this could be shoved over to the dev 2.1 branch and looked at there while the bugfix branch here is left with what we know works.

Reply:
When behavior is anomalous I am not satisfied just to do a workaround and move on. It is important to understand the underlying causes and provide patches that are comprehensive. We are using this methodology in several other places, and we need to understand the pitfalls. Since you are busy I don't mind trying to understand the problem. If I can't reproduce the behavior you'll be on the hook to re-test under the same conditions where you saw the issue.

Reply:
I saw a tweak to the remember macro shortly behind this pr being opened that I could check at some point. I know endstops use this as well. Regardless for something that  contains movement I still don't like not handling it explicitly. Regardless, I've kept this in the Creality crx and formbot Trex branches I maintain so I can minimize affected users.

Reply:
By all means, do what works best and represents least hassle. As I focus purely on cleanup and bug-fixes over the rest of the month this is one of the things that will get closer scrutiny.